src/app/_components/product.tsx

```diff
@@ -0,0 +1,221 @@
+            Imprimir presupuesto
+          </p>
+        </div>
+      </div>
+    </div>
+  );
+
+  const Mobile = () => (
+    <div>
+      <Image
+        src="/img/flipa-box.svg"
+        priority
+        alt=""
+        height={400}
+        width={400}
+        objectFit="cover"
+        className="w-full"
+      />
+      <div className="flex flex-col gap-4 p-4">
+        <div className="mt-4 flex justify-between">
+          <div>
+            <h2 className="text-2xl text-customGray">{restaurant}</h2>
+            <h1 className="bg-custom-gradient bg-clip-text text-3xl text-transparent">
+              {name}
+            </h1>
+          </div>
+        </div>
+
+        <div className="flex items-center gap-7">
+          <div className="flex items-center gap-1">
+            <div className="-mt-[0.15rem] h-4 w-4 bg-[url('/card/star-product.svg')]"></div>
+            <p className="cursor-pointer bg-custom-gradient bg-clip-text text-sm">
+              {calification}
+            </p>
+          </div>
+          <p className="bg-clip-text text-sm text-blue-700">
+            Ver calificaciones
+          </p>
+        </div>
+
+        <div className=" text-gray-700">
+          <h1 className="text-xl">Descripción</h1>
+          <div className="text-base text-gray-700">
+            {firstItem}
+            <ul className="list-inside list-disc">
+              {restItems.map((item, index) => (
+                <li key={index}>{item}</li>
+              ))}
+            </ul>
+          </div>
+        </div>
+
+        <div className="mb-10 mt-10 flex flex-row bg-custom-gradient bg-clip-text text-2xl font-medium text-transparent">
+          <div>
+            ${price}
+            <span className="text-text">/ unidad</span>
+          </div>
+        </div>
+
+        <div className="flex flex-row gap-8">
+          <div className="mb-8 flex gap-0">
+            <button
+              onClick={decreaseQuantity}
+              className="rounded-l-md bg-gray-200 p-2 hover:bg-gray-300 focus:outline-none focus:ring-2 focus:ring-gray-400 focus:ring-opacity-50"
+            >
+              -
+            </button>
+            <div className="flex w-12 items-center justify-center border-y border-gray-200 text-center">
+              {quantity}
+            </div>
+            <button
+              onClick={increaseQuantity}
+              className="rounded-r-md bg-gray-200 p-2 text-black hover:bg-gray-300 focus:outline-none focus:ring-2 focus:ring-gray-400 focus:ring-opacity-50"
+            >
+              +
+            </button>
+          </div>
+          <Button className={buttonVariants({ variant: "primary" })}>
+            Comprar
+          </Button>
+        </div>
+      </div>
+    </div>
+  );
+
+  if (variant === "mobile") {
+    return Mobile();
+  } else {
+    return Default();
+  }
```

Comment:
Is this the standard way to do it? Just asking

Reply:
Nice call. Since we were calling 2 instances of the Product component for the different screen sizes when the screen changed, the component was unmounting while the other was mounting, resulting on the state of the Product component to reset. So the total quantity was reverting to 0 if the user where to change its screen size. With the new fix, we are creating a single instance of Product and just rendering the view using tailwindcss hidden display properties.